app/main.py

```diff
@@ -1,54 +1,36 @@
-import json
-import xml.etree.ElementTree as ET
-
-
-class Book:
-    def __init__(self, title: str, content: str):
-        self.title = title
-        self.content = content
-
-    def display(self, display_type: str) -> None:
-        if display_type == "console":
-            print(self.content)
-        elif display_type == "reverse":
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown display type: {display_type}")
-
-    def print_book(self, print_type: str) -> None:
-        if print_type == "console":
-            print(f"Printing the book: {self.title}...")
-            print(self.content)
-        elif print_type == "reverse":
-            print(f"Printing the book in reverse: {self.title}...")
-            print(self.content[::-1])
-        else:
-            raise ValueError(f"Unknown print type: {print_type}")
-
-    def serialize(self, serialize_type: str) -> str:
-        if serialize_type == "json":
-            return json.dumps({"title": self.title, "content": self.content})
-        elif serialize_type == "xml":
-            root = ET.Element("book")
-            title = ET.SubElement(root, "title")
-            title.text = self.title
-            content = ET.SubElement(root, "content")
-            content.text = self.content
-            return ET.tostring(root, encoding="unicode")
-        else:
-            raise ValueError(f"Unknown serialize type: {serialize_type}")
+from app.book import Book
+from app.display import ConsoleDisplayer, ReverseDisplayer
+from app.print import ConsolePrinter, ReversePrinter
+from app.serializer import JsonSerializer, XmlSerializer
 
 
 def main(book: Book, commands: list[tuple[str, str]]) -> None | str:
     for cmd, method_type in commands:
         if cmd == "display":
-            book.display(method_type)
+            if method_type == "console":
+                ConsoleDisplayer.display(book)
+            elif method_type == "reverse":
+                ReverseDisplayer.display(book)
+            else:
+                raise ValueError(f"Unknown display type: {method_type}")
         elif cmd == "print":
-            book.print_book(method_type)
+            if method_type == "console":
+                ConsolePrinter.print_book(book)
+            elif method_type == "reverse":
+                ReversePrinter.print_book(book)
+            else:
+                raise ValueError(f"Unknown print type: {method_type}")
         elif cmd == "serialize":
-            return book.serialize(method_type)
+            if method_type == "json":
+                return JsonSerializer.serialize(book)
+            elif method_type == "xml":
+                return XmlSerializer.serialize(book)
+            else:
+                raise ValueError(f"Unknown serialize type: {method_type}")
+        else:
+            raise ValueError(f"Unknown cmd command: {cmd}")
```

Comment:
In case where we talking about SOLID you should adhere SOLID principles, In your case, you could create separate classes for handling serialization, printing, and displaying(and handler with specific fetching properties). Also you should create separate interfaces for different types of handler which will handle somehow data related to book.

And these entities should get all reldated data on main process.